stl/inc/any

```diff
@@ -168,8 +168,10 @@ public:
     }
 
     any& operator=(any&& _That) noexcept {
-        reset();
-        _Move_from(_That);
+        if (this != &_That) {
+            reset();
+            _Move_from(_That);
+        }

```

Comment:
I think this "fix" doesn't essentially resolve the problem. For example:
```
    any a = std::make_any<any>(1); // a has an `any(1)`
    a = any_cast<any&&>(a);// ugh...
```
Though risking higher performance impact, I think we have to move-construct a local any then `_Move_from` it to do the real fix...